pebble/src/main/java/com/mitchellbosecke/pebble/attributes/MethodAccessValidator.java

```diff
@@ -0,0 +1,46 @@
+package com.mitchellbosecke.pebble.attributes;
+
+import com.mitchellbosecke.pebble.error.ClassAccessException;
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Method;
+
+public class MethodAccessValidator {
+
+  private static final String[] FORBIDDEN_OBJECT_CLASS_METHODS = {"getClass",
+      "wait",
+      "notify",
+      "notifyAll"};
+
+  void checkIfAccessIsAllowed(Object object, Method member, String filename, int lineNumber) {
+    if (this.isAccessUnsafe(object, member)) {
+      throw new ClassAccessException(member, filename, lineNumber);
+    }
+  }
+
+  private boolean isAccessUnsafe(Object object, Method member) {
+    return object instanceof Class
+        || object instanceof Runtime
+        || object instanceof Thread
+        || object instanceof ThreadGroup
+        || object instanceof System
+        || object instanceof AccessibleObject
```

Comment:
All those classes can only be defined in the JDK. Just wondering if if would speeding things to use class identity equality:

```java
Class<?> clazz = object.getClass();
return clazz == Class.class || clazz == Runtime.class etc
```

Reply:
Performance seems to be the same when running with https://github.com/PebbleTemplates/pebble-performance-test

But if I use this method, I need to add some reflect class to the list like this
```
Class<?> clazz = object.getClass();
    boolean methodForbidden = clazz == Class.class
        || clazz == Runtime.class
        || clazz == Thread.class
        || clazz == ThreadGroup.class
        || clazz == System.class
        || clazz == AccessibleObject.class
        || clazz == Method.class
        || clazz == Field.class
        || clazz == Constructor.class
        || this.isUnsafeMethod(method);
```